File: apprefurbished/src/main/java/com/androidcollider/easyfin/transaction_categories/nested/TransactionCategoriesNestedModule.java
package com.androidcollider.easyfin.transaction_categories.nested;

import android.content.Context;

import com.androidcollider.easyfin.common.managers.resources.ResourcesManager;
import com.androidcollider.easyfin.common.repository.Repository;

import dagger.Module;
import dagger.Provides;

/**
 * @author Ihor Bilous
 */

@Module
public class TransactionCategoriesNestedModule {

    @Provides
    TransactionCategoriesNestedMVP.Model provideTransactionCategoriesNestedMVPModel(Repository repository) {
        return new TransactionCategoriesNestedModel(repository);
    }

    @Provides
    TransactionCategoriesNestedMVP.Presenter provideTransactionCategoriesNestedMVPPresenter(Context context,
                                                                                            TransactionCategoriesNestedMVP.Model model,
                                                                                            ResourcesManager resourcesManager) {
        return new TransactionCategoriesNestedPresenter(context, model, resourcesManager);
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/transaction_categories/nested/TransactionCategoriesNestedModule.kt
package com.androidcollider.easyfin.transaction_categories.nested

import android.content.Context
import com.androidcollider.easyfin.common.managers.resources.ResourcesManager
import com.androidcollider.easyfin.common.repository.Repository
import dagger.Module
import dagger.Provides

/**
 * @author Ihor Bilous
 */
@Module
class TransactionCategoriesNestedModule {
    @Provides
    fun provideTransactionCategoriesNestedMVPModel(repository: Repository?): TransactionCategoriesNestedMVP.Model {
        return TransactionCategoriesNestedModel(repository)
    }

    @Provides
    fun provideTransactionCategoriesNestedMVPPresenter(
        context: Context?,
        model: TransactionCategoriesNestedMVP.Model?,
        resourcesManager: ResourcesManager?
    ): TransactionCategoriesNestedMVP.Presenter {
        return TransactionCategoriesNestedPresenter(context, model, resourcesManager)
    }
}